classes/submission/reviewAssignment/Repository.php

```diff
@@ -0,0 +1,256 @@
+    }
+
+    /**
+     * Add a new review assignment
+     */
+    public function add(ReviewAssignment $reviewAssignment): int
+    {
+        $id = $this->dao->insert($reviewAssignment);
+        $reviewAssignment->stampModified();
+        Hook::call('ReviewAssignment::add', [$reviewAssignment]);
+        $this->updateReviewRoundStatus($reviewAssignment);
+
+        return $id;
+    }
+
+    /**
+     * Edit a review assignment
+     */
+    public function edit(ReviewAssignment $reviewAssignment, array $params)
+    {
+        $newReviewAssignment = clone $reviewAssignment;
+        $newReviewAssignment->setAllData(array_merge($newReviewAssignment->_data, $params));
+        $newReviewAssignment->stampModified();
+
+        Hook::call('ReviewAssignment::edit', [$newReviewAssignment, $reviewAssignment, $params]);
+
+        $this->dao->update($newReviewAssignment);
+        $this->updateReviewRoundStatus($newReviewAssignment);
+    }
+
+    /**
+     * Delete a review assignment
+     */
+    public function delete(ReviewAssignment $reviewAssignment)
+    {
+        Hook::call('ReviewAssignment::delete::before', [$reviewAssignment]);
+
+        $reviewFormResponseDao = DAORegistry::getDAO('ReviewFormResponseDAO'); /** @var ReviewFormResponseDAO $reviewFormResponseDao */
+        $reviewFormResponseDao->deleteByReviewId($reviewAssignment->getId());
+
+        $reviewFilesDao = DAORegistry::getDAO('ReviewFilesDAO'); /** @var ReviewFilesDAO $reviewFilesDao */
+        $reviewFilesDao->revokeByReviewId($reviewAssignment->getId());
+
+        $notificationDao = DAORegistry::getDAO('NotificationDAO'); /** @var NotificationDAO $notificationDao */
+        $notificationDao->deleteByAssoc(Application::ASSOC_TYPE_REVIEW_ASSIGNMENT, $reviewAssignment->getId());
+
+        $this->dao->delete($reviewAssignment);
+
+        $this->updateReviewRoundStatus($reviewAssignment);
+
+        Hook::call('ReviewAssignment::delete', [$reviewAssignment]);
+    }
+
+    /**
+     * Delete a collection of announcements
+     */
+    public function deleteMany(Collector $collector)
+    {
+        foreach ($collector->getMany() as $reviewAssignment) {
+            $this->delete($reviewAssignment);
+        }
+    }
+
+    /**
+     * Return the review methods translation keys.
+     */
+    public function getReviewMethodsTranslationKeys(): array
+    {
+        return [
+            ReviewAssignment::SUBMISSION_REVIEW_METHOD_DOUBLEANONYMOUS => 'editor.submissionReview.doubleAnonymous',
+            ReviewAssignment::SUBMISSION_REVIEW_METHOD_ANONYMOUS => 'editor.submissionReview.anonymous',
+            ReviewAssignment::SUBMISSION_REVIEW_METHOD_OPEN => 'editor.submissionReview.open',
+        ];
+    }
+
+    /**
+     * Update the status of the review round an assignment is attached to. This
+     * should be fired whenever a reviewer assignment is modified.
+     */
+    protected function updateReviewRoundStatus(ReviewAssignment $reviewAssignment): bool
```

Comment:
I suspect functions like this would be better placed in the DAO, but since it's protected, no biggie if we decide to rearrange things later.